types/figma/index.d.ts

```diff
@@ -0,0 +1,666 @@
+// Type definitions for Figma API v1.0.0
+// Project: https://www.figma.com/plugin-docs/intro
+// Definitions by: Figma <https://www.figma.com>
+//                 Rudi Chen <rudi@figma.com>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+//
+// Talk to the Figma team regarding changes to the typings. While
+// changes to the types aren't breaking changes to how plugins run,
+// it is preferrable to coordinate break type changes with breaking
+// API changes.
+
+// Global variable with Figma's plugin API.
+declare const figma: PluginAPI
+declare const __html__: string
+
+interface PluginAPI {
+    readonly apiVersion: "1.0.0"
+    readonly command: string
+    readonly root: DocumentNode
+    readonly viewport: ViewportAPI
+    closePlugin(message?: string): void
+
+    showUI(html: string, options?: ShowUIOptions): void
+    readonly ui: UIAPI
+
+    readonly clientStorage: ClientStorageAPI
+
+    getNodeById(id: string): BaseNode | null
+    getStyleById(id: string): BaseStyle | null
+
+    currentPage: PageNode
+
+    readonly mixed: symbol
+
+    createRectangle(): RectangleNode
+    createLine(): LineNode
+    createEllipse(): EllipseNode
+    createPolygon(): PolygonNode
+    createStar(): StarNode
+    createVector(): VectorNode
+    createText(): TextNode
+    createBooleanOperation(): BooleanOperationNode
+    createFrame(): FrameNode
+    createComponent(): ComponentNode
+    createPage(): PageNode
+    createSlice(): SliceNode
+
+    createPaintStyle(): PaintStyle
+    createTextStyle(): TextStyle
+    createEffectStyle(): EffectStyle
+    createGridStyle(): GridStyle
+
+    importComponentByKeyAsync(key: string): Promise<ComponentNode>
+    importStyleByKeyAsync(key: string): Promise<BaseStyle>
+
+    listAvailableFontsAsync(): Promise<Font[]>
+    loadFontAsync(fontName: FontName): Promise<void>
+    readonly hasMissingFont: boolean
+
+    createNodeFromSvg(svg: string): FrameNode
+
+    createImage(data: Uint8Array): Image
+    getImageByHash(hash: string): Image
+
+    group(nodes: ReadonlyArray<BaseNode>, parent: BaseNode & ChildrenMixin, index?: number): FrameNode
+    flatten(nodes: ReadonlyArray<BaseNode>, parent?: BaseNode & ChildrenMixin, index?: number): VectorNode
+}
+
+interface ClientStorageAPI {
+    getAsync(key: string): Promise<any | undefined>
```

Comment:
`unknown` is better than `any | undefined` here, unless you want to explicitly exclude `null`, in which case `{} | undefined` is the shortest type you could write. But `object | string | number | boolean | symbol | bigint | undefined` might be easier to read.